mp/src/game/shared/momentum/mom_rocket.cpp

```diff
@@ -299,7 +349,13 @@ CMomRocket *CMomRocket::EmitRocket(const Vector &vecOrigin, const QAngle &vecAng
     pRocket->SetRadius(146.0f);
 
     pRocket->CreateSmokeTrail();
-    pRocket->EmitSound("Missile.Ignite");
+
+    static ConVarRef mom_rj_sounds("mom_rj_sounds");
+    if ((!mom_rj_rocketmodel.GetBool() && !mom_rj_sounds.GetInt() == 0) ||
+        (!mom_rj_rocketmodel.GetBool() && mom_rj_sounds.GetInt() == 2))
```

Comment:
What exactly is this checking? `mom_rj_sounds.GetInt()` is negated and then checked to be `== 0`, could be a bug.

Use some const local bools to describe each of these conditions, this is really hard to follow as it is.

Reply:
This is meant to only play the sound of the rocket traveling (the HL2 default one) if TF2 sounds are not on (since TF2 rockets don't have a travel sound), the TF2 rocket model is not used and mom_rj_sounds is not 0 (i.e. sounds are not disabled). I will try to make it more descriptive.

Also I think I accidentally negated `mom_rj_sounds.GetInt()` instead of negating the operator.